sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,312 @@
+    // we're compensating with this.
+    const fakeThis: any = {
+      ...this,
+      currentTest: {
+        file: "test/recorder.browser.spec.ts",
+        // The hash in our expected recording is made out of an empty function.
+        // This function has something inside, which means it has changed.
+        fn: getAnotherNoOpFunction()
+      }
+    };
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    // Restoring the XHR, otherwise we can't test this.
+    window.XMLHttpRequest = originalXHR;
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    xhrMock.teardown();
+    recorder.stop();
+    console.log = originalConsoleLog;
+
+    // Now we check the hash has changed in the recorded console.log output.
+
+    // Here we confirm that the recorder generated an expected output on the console.logs.
+    // This output is used to generate the recording files in the filesystem, though here we're only
+    // checking what was that the recorded emitted to the standard output.
+    expect(savedConsoleLogParams[0][0]).to.equal(
+      // The recordings here are empty because we hijacked the XHR.
+      // See the playback test for an example of a properly constructed recording object.
+      // TODO: Find a way to capture the complete output.
+      JSON.stringify({
+        writeFile: true,
+        path:
+          "./recordings/browsers/the_recorders_public_api_on_a_browser/recording_softrecord_should_rerecord_a_simple_outdated_test.json",
+        content: {
+          recordings: [],
+          uniqueTestInfo: {
+            uniqueName: {},
+            newDate: {}
+          },
+          hash: MD5(getAnotherNoOpFunction().toString())
+        }
+      })
+    );
+  });
+
+  it("soft-record should skip a simple unchanged test", async function() {
+    // Setting up the playback mode.
+    // The PATH environment variable is not needed on playback.
+    (window as any).__env__.TEST_MODE = "soft-record";
+
+    // This is to emulate what 'karma-json-preprocessor' does for us during the real scenarios.
+    (window as any).__json__ = {
+      ["recordings/browsers/the_recorders_public_api_on_a_browser/recording_softrecord_should_skip_a_simple_unchanged_test.json"]: {
+        recordings: [
+          {
+            method: "GET",
+            url: "/replaced",
+            response: expectedHttpResponse
+          }
+        ],
+        uniqueTestInfo: { uniqueName: {}, newDate: {} },
+        // This is the expected hash
+        hash: MD5(getNoOpFunction().toString())
+      }
+    };
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    (this as any).currentTest = {
```

Comment:
do we need to override `currentTest` on `this`, or can we just set it directly inside `fakeThis` below?

Reply:
I should be able to set it on fakeThis. I thought I did it everywhere, but I missed some things 😁